engine/src/ast/function_expr.rs

```diff
@@ -182,9 +182,9 @@ impl<'i, 's> LexWith<'i, &'s Scheme> for FunctionCallExpr<'s> {
                     rest
                 }
                 _ => {
-                    function_call
-                        .args
-                        .push(FunctionCallArgExpr::Literal(opt_arg.default_value.clone()));
+                    function_call.args.push(FunctionCallArgExpr::Literal(
+                        (&opt_arg.default_value).into(),
```

Comment:
As discussed, let's move this into the execution stage instead.

Currently, you have `LhsValue`, convert `&LhsValue` into `RhsValue` and then during execution you convert it back to `LhsValue` again.

If you move it to the execution, then you can take `LhsValue` and pass to the function pretty much as-is (but perhaps need to use same `::Borrowed` trick as you already use in the other place).